### PR TITLE
add metric per variant, add warnings for pyvcf bugs

### PR DESCRIPTION
**Pull request by CGUTA**
Errors
885854
('TACAGGTACAC|TACAGGTACAC', 'TACAC|TACAC') Nonsense match #1
947926
('C|C', 'CT|CT') Nonsense match #2

Test ref and check
1	947926	.	C	G	.	PASS # returned record
1	947926	.	CT	C	.	PASS

1	947926	.	C	G	.	PASS # 1st request
1	947926	.	CT	C	.	PASS # 2nd request

This error appears when the position 947926 is requested from check to ref the second time. Aparently fetch gets the first instance of the position. Because there is no way of telling which is which this error is unavoidable at the moment

Test ref and check
1	885854	.	TACAC	T	.	PASS
1	885854	.	TACAC	T	.	PASS
Test ref and check
1	885848	.	TACAGGTACAC	T	.	PASS
1	885848	.	TACAGGTACAC	T	.	PASS

This error ocurrs due to an incorrect fetching protocol and thus it is a bug in Pyvcf.
Below the code that reproduces the error with the test files.

import vcf
import pysam

ref_reader = vcf.Reader(filename='testRef.vcf.gz')
chk_reader = vcf.Reader(filename='testCheck.vcf.gz')
refsnp = ref_reader.fetch(1, 885854)
chksnp = chk_reader.fetch(1, 885854)
print refsnp
print chksnp # positions dont match
refsnp = ref_reader.fetch(1, 885848)
chksnp = chk_reader.fetch(1, 885848)
print refsnp
print chksnp # positions match

Normal usage 
python phasingQC.py -i testRef.vcf.gz -I testCheck.vcf.gz -o testOutput.txt

### CODE REVIEW

File: phasingQC/phasingQC_0.02.py [hunk 1]
```diff
@@ -105,15 +105,25 @@ def mismatch_type(ref_gt, eva_gt):
 #print(links)
 ## where do the closes go?
 
-
 metric = [[0 for i in range(7)] for j in  range(len(checkSamples))]
 rev = [False for i in range(len(checkSamples))]
 
+output2 = open('metric_per_snp', "w" )
```
Comment (npklein) [inline phasingQC/phasingQC_0.02.py]: output2? what is output1?

File: phasingQC/phasingQC_0.02.py [hunk 2]
```diff
@@ -131,20 +141,38 @@ def mismatch_type(ref_gt, eva_gt):
...
         else:
             metric[sampleCounter][0] += 1
+            metric_per_snp[0] += 1
         sampleCounter += 1
-else:
-    print("Nothing was skipped :)")
+    ### Prints a metric per SNPS to detect mismatches that ocurr globally... techical probbly
+    previousPosition = record.POS
+    output2.write(str(record.CHROM))
+    output2.write("\t")
```
Comment (npklein) [inline phasingQC/phasingQC_0.02.py]: better to use
with open('metric_per_snp.txt','w') as output:
    output.write(...)
this gets automatically closed outside the 'with' scope

All the output.write() can also go into one, e.g.
      output2.write(str(record.CHROM) + '\t' +
                            str(record.POS) + '\t' + etc
                            )

**Comment by CGUTA** [comment]
The with open as output : I implemented in the second output.. The first one would get overwritten if i did just around where the writes are...... and indenting the whole thing does not look pretty. Should I do it anyway?

**Comment by npklein** [comment]
@CGUTA need to make a fresh pull request after deleting the history in the way I emailed yesterday

**Pull request closed without merging**